core/citrus-base/src/test/java/com/consol/citrus/context/TestContextTest.java

```diff
@@ -173,6 +191,13 @@ public void testUnknownFromPathExpression() {
         } catch (CitrusRuntimeException e) {
             Assert.assertEquals(e.getMessage(), "Unknown variable 'something.else'");
         }
+        
+        try {
+            context.getVariable("${helloData[1]}");
+            Assert.fail("Missing exception due to indexed access to non array variable");
+        } catch (CitrusRuntimeException e) {
+            Assert.assertTrue(e.getMessage().endsWith(""));
+        }
```

Comment:
`assertThrows` would probably be an option here.

Reply:
Just following the pattern used in this TestCase.